### PR TITLE
topic, key and partition changes

### PR DESCRIPTION
**Pull request by vanchi-zendesk**
1. Write to a user provided topic (default "maxwell")
2. Also write the db name for the event.
3. Use a key : dbname/tablename
4. Write a partition based on the hash of the db-name.

@zendesk/rules

### CODE REVIEW

**Comment by osheroff** [comment]
yokay; if we're leaving it up to the user to create the kafka partition, let's put a note in the docs to create  at least nDatabases partitions.

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public void onCompletion(RecordMetadata md, Exception e) {
 			try {
 				if ( LOGGER.isDebugEnabled()) {
 					LOGGER.debug("-> " + md.topic() + ":" + md.offset());
+					LOGGER.debug("    partition:" + md.partition());
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: can be moved to the line above.

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java [hunk 2]
```diff
@@ -46,24 +47,43 @@ public void onCompletion(RecordMetadata md, Exception e) {
 public class MaxwellKafkaProducer extends AbstractProducer {
 	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellKafkaProducer.class);
 	private final KafkaProducer<byte[], byte[]> kafka;
+	private final String topic;
+	private final int numPartitions;
 
 	public MaxwellKafkaProducer(MaxwellConfig config, Properties kafkaProperties) {
 		super(config);
 
+		LOGGER.debug(kafkaProperties.toString());
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: don't mind you leaving in the debug line, but give it some context maybe?

**Comment by osheroff** [comment]
nice job, :+1:, looking forward to seeing how this performs in prod!

**Comment by vanchi-zendesk** [comment]
@osheroff Do check the last commit. It rebuilds the site. (You might want to checkout locally and do mkdocs serve to see the changes)

File: docs/docs/config.md [hunk 3]
```diff
@@ -26,19 +27,31 @@ password=PASSWORD             | mysql password
 host=HOST                     | mysql host
 port=PORT                     | mysql port
 producer=stdout,kafka,        | where shall we send these rows, sir?
-kafka.*=XXX                   | any options prefixed with 'kafka.' will be passed into the kafka producer library
+kafka.maxwell.topic           | provide a topic for maxwell to write to. Default will be "maxwell".
```
Comment (osheroff) [inline docs/docs/config.md]: maybe let's give kafka the `kafka.` namespace entirely, make this "kafa_topic" instead...

File: docs/docs/config.md [hunk 4]
```diff
@@ -26,19 +27,31 @@ password=PASSWORD             | mysql password
...
+Any options in the configuration file prefixed with `kafka.` have that prefix stripped off, and passed directly
 into the "new producer" configuration, as described here: [http://kafka.apache.org/documentation.html#newproducerconfigs](http://kafka.apache.org/documentation.html#newproducerconfigs)
 
-Maxwell sets the following options by default, but you can override them in config.properties.
+Maxwell sets the following options by default, but you can override them in `config.properties`.
 
 - (kafka.)acks = 1
 - (kafka.)compression.type = gzip
 
+### Topic and partitioning
```
Comment (osheroff) [inline docs/docs/config.md]: let's move this to a new documentation section on kafka.

File: docs/docs/config.md (hunk 3 above)
Comment (vanchi-zendesk) [inline docs/docs/config.md]: Agreed.

File: docs/docs/config.md (hunk 4 above)
Comment (vanchi-zendesk) [inline docs/docs/config.md]: done.

**Comment by vanchi-zendesk** [comment]
updated with changes.

**Comment by osheroff** [comment]
:+1:

**Pull request merged**